test/authorization/service/role.service.test.ts

```diff
@@ -78,170 +95,226 @@ describe('test Role Service', () => {
+      findOneByMock.mockResolvedValue(roles[0]);
+
+      const result = await roleService.getRoleById(VALID_ROLE_ID);
+
+      expect(findOneByMock).toBeCalledWith({
+        id: VALID_ROLE_ID,
+      });
+
+      expect(result).toEqual(roles[0]);
+    });
+
+    it('should throw exception for invalid id', async () => {
+      let result;
+      findOneByMock.mockResolvedValue(null);
 
-    rolesRepository.insert(roles[0]).returns(Arg.any());
+      try {
+        result = await roleService.getRoleById(INVALID_ROLE_ID);
+      } catch (error) {
+        expect(findOneByMock).toBeCalledWith({
+          id: INVALID_ROLE_ID,
+        });
 
-    const resp = await roleService.createRole(input);
-    expect(resp).toEqual(roles[0]);
+        expect(error).toBeInstanceOf(RoleNotFoundException);
+      }
+    });
   });
 
-  it('should update a role', async () => {
-    const input: UpdateRoleInput = {
-      name: 'Test1',
-    };
+  describe('createRole', () => {
+    it('should create a role', async () => {
+      const input: NewRoleInput = {
+        name: 'Test1',
+      };
+
+      saveMock.mockResolvedValue(roles[0]);
+
+      const result = await roleService.createRole(input);
 
-    rolesRepository
-      .update('ae032b1b-cc3c-4e44-9197-276ca877a7f8', input)
-      .returns(Promise.resolve(Arg.any()));
+      expect(saveMock).toBeCalledWith(input);
 
-    const resp = await roleService.updateRole(
-      'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
-      input,
-    );
-    expect(resp).toEqual(roles[0]);
+      expect(result).toEqual(roles[0]);
+    });
   });
 
-  it('should add permissions to a role', async () => {
-    const request = [
-      {
-        roleId: 'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
-        permissionId: '2b33268a-7ff5-4cac-a87a-6bfc4430d34c',
-      },
-    ];
-    const input: UpdateRolePermissionInput = {
-      permissions: ['2b33268a-7ff5-4cac-a87a-6bfc4430d34c'],
-    };
-
-    permissionRepository
-      .findByIds(['2b33268a-7ff5-4cac-a87a-6bfc4430d34c'])
-      .resolves(permissions);
-
-    rolePermissionRepository.create(request).returns(request);
-
-    permissionRepository
-      .createQueryBuilder('permission')
-      .returns(permissionQueryBuilder);
-    permissionQueryBuilder
-      .leftJoinAndSelect(
-        RolePermission,
-        'rolePermission',
-        'permission.id = rolePermission.permissionId',
-      )
-      .returns(permissionQueryBuilder);
-
-    permissionQueryBuilder
-      .where('rolePermission.roleId = :roleId', {
-        roleId: 'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
-      })
-      .returns(permissionQueryBuilder);
-
-    permissionQueryBuilder.getMany().resolves(permissions);
-
-    connectionMock.transaction(Arg.any()).resolves(request);
-
-    const resp = await roleService.updateRolePermissions(
-      'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
-      input,
-    );
-    expect(resp).toEqual(permissions);
+  describe('updateRole', () => {
```

Comment:
Test fails.

Reply:
Will fix service related tests in a separate PR.